src/main/cli/cli.c

```diff
@@ -6121,120 +6220,169 @@ static void cliHelp(char *cmdline)
-                        break;
-                    if (!pstart->name[bufferIndex] && bufferIndex < sizeof(cliBuffer) - 2) {
-                        /* Unambiguous -- append a space */
-                        cliBuffer[bufferIndex++] = ' ';
-                        cliBuffer[bufferIndex] = '\0';
-                        break;
-                    }
-                    cliBuffer[bufferIndex] = pstart->name[bufferIndex];
+                if ((options = checkCommand(cliBuffer, cmd->name))) {
+                    break;
                 }
             }
-            if (!bufferIndex || pstart != pend) {
-                /* Print list of ambiguous matches */
-                cliPrint("\r\033[K");
-                for (cmd = pstart; cmd <= pend; cmd++) {
-                    cliPrint(cmd->name);
-                    cliWrite('\t');
-                }
-                cliPrompt();
-                i = 0;    /* Redraw prompt */
+            if (cmd < cmdTable + ARRAYLEN(cmdTable)) {
+                cmd->func(options);
+            } else {
+                cliPrintError("UNKNOWN COMMAND, TRY 'HELP'");
             }
-            for (; i < bufferIndex; i++)
-                cliWrite(cliBuffer[i]);
-        } else if (!bufferIndex && c == 4) {   // CTRL-D
-            cliExit(cliBuffer);
+            bufferIndex = 0;
+        }
+
+        memset(cliBuffer, 0, sizeof(cliBuffer));
+
+        // 'exit' will reset this flag, so we don't need to print prompt again
+        if (!cliMode) {
             return;
-        } else if (c == 12) {                  // NewPage / CTRL-L
-            // clear screen
-            cliPrint("\033[2J\033[1;1H");
-            cliPrompt();
-        } else if (bufferIndex && (c == '\n' || c == '\r')) {
-            // enter pressed
-            cliPrintLinefeed();
+        }
 
-            // Strip comment starting with # from line
-            char *p = cliBuffer;
-            p = strchr(p, '#');
-            if (NULL != p) {
-                bufferIndex = (uint32_t)(p - cliBuffer);
-            }
+        cliPrompt();
+    } else if (c == 127) {
+        // backspace
+        if (bufferIndex) {
+            cliBuffer[--bufferIndex] = 0;
+            cliPrint("\010 \010");
+        }
+    } else if (bufferIndex < sizeof(cliBuffer) && c >= 32 && c <= 126) {
+        if (!bufferIndex && c == ' ')
+            return; // Ignore leading spaces
+        cliBuffer[bufferIndex++] = c;
+        cliWrite(c);
+    }
+}
 
-            // Strip trailing whitespace
-            while (bufferIndex > 0 && cliBuffer[bufferIndex - 1] == ' ') {
-                bufferIndex--;
-            }
+void cliProcess(void)
+{
+    if (!cliWriter) {
+        return;
+    }
 
-            // Process non-empty lines
-            if (bufferIndex > 0) {
-                cliBuffer[bufferIndex] = 0; // null terminate
+    // Be a little bit tricky.  Flush the last inputs buffer, if any.
```

Comment:
not your code/comment, but I don't know what 'Be a little bit tricky.' means here, can the comment be improved?

Reply:
`// Clear out the input buffer before we start listening for CLI input`
or
`// Flush the input buffer before we start listening for CLI input`

The best guess I have is that they had lots of trouble with the CLI code not randomly not processing the first command until they thought to flush the buffer.

https://github.com/betaflight/betaflight/commit/cf7c9d41000f52d86594d5eadf21d8f81109d713